build_baselines.py

```diff
@@ -207,7 +207,15 @@ def build_base(iso, md5, replace_existing, vmServer=None, prependString = "", in
                     })
 
     packerfile = os.path.join(temp_path, "current_packer.json")
-    packer_obj.save_config(packerfile)
+
+    packer_obj.save_config(packerfile)  
+    
+    if factory_image:
+        # Software we don't want on our factory image
+        software = ["python", "java", "ruby"]
+        for provisioner in packer_obj.local_packer['provisioners']:
+            if any(x in str(provisioner.values()) for x in software):
+                packer_obj.local_packer['provisioners'].remove(provisioner)
```

Comment:
```suggestion

    if factory_image:
        # Software we don't want on our factory image
        software = ["python", "java", "ruby"]
        for provisioner in packer_obj.local_packer['provisioners']:
            if any(x in str(provisioner.values()) for x in software):
                provisioner['except'] = only

    packer_obj.save_config(packerfile)
```

The manipulation of provisioners need to be complete before `packer_obj.save_config()` is called to write the packer file to disk.

Also removing from a dictionary in python can change the order of objects.  Since the provisioners need to be in order, simply [add an `except` key](https://www.packer.io/docs/templates/provisioners#run-on-specific-builds) to the provisioner to be suppressed using the builder type being executed as one to suppress the provisioner for.  In this case `only` is already defined as an array of builders to be executed so it makes for an easy value to insert.